src/context.rs

```diff
@@ -50,11 +51,13 @@ impl<'a> Context<'a> {
                 })
             });
 
-        Context::new_with_dir(arguments, path)
+        let logical_path: Option<PathBuf> = arguments.value_of("logical-path").map(From::from);
+
+        Context::new_with_dir(arguments, path, logical_path)
     }
 
     /// Create a new instance of Context for the provided directory
-    pub fn new_with_dir<T>(arguments: ArgMatches, dir: T) -> Context
+    pub fn new_with_dir<T>(arguments: ArgMatches, path: T, logical_path: Option<PathBuf>) -> Context
```

Comment:
**suggestion:**
Since we already have the `T: Into<PathBuf>` here, it seems sensible to use it for `logical_path` too:
```suggestion
    pub fn new_with_dir<T>(arguments: ArgMatches, path: T, logical_path: Option<T>) -> Context
```